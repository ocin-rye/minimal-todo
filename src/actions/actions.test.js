import configureMockStore from 'redux-mock-store';
import { addTodo, deleteTodo, toggleTodoCompletion, setTodoListFilter } from './index';

// Create a mock store
const mockStore = configureMockStore();
const store = mockStore({});

describe('action creators', () => {
  const todo = {
    completed: false,
    id: '1UpbCPrF6mODLiVE6Uw7-',
    text: 'Sample Todo Text.',
  };

  beforeEach(() => {
    store.clearActions();
  });

  it('creates an ADD_TODO action', () => {
    // Dispatch the addTodo action with the values of a new todo.
    store.dispatch(addTodo('This is text todo text.', 'testTodoId'));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('creates a DELETE_TODO action', () => {
    // Dispatch the deleteTodo action with the values of an existing todo.
    store.dispatch(deleteTodo(todo));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('creates an TOGGLE_TODO_COMPLETION action', () => {
    // Dispatch the toggleTodoCompletion action with the values of an existing todo.
    store.dispatch(toggleTodoCompletion(todo));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('creates an SET_TODO_LIST_FILTER action', () => {
    // Dispatch the setTodoListFilter action with the value of a todo list filter setting.
    const filterSetting = 'ALL';
    store.dispatch(setTodoListFilter(filterSetting));
    expect(store.getActions()).toMatchSnapshot();
  });
});
