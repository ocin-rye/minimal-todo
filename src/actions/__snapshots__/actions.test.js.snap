// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action creators creates a DELETE_TODO action 1`] = `
Array [
  Object {
    "payload": Object {
      "completed": false,
      "id": "1UpbCPrF6mODLiVE6Uw7-",
      "text": "Sample Todo Text.",
    },
    "type": "delete_todo",
  },
]
`;

exports[`action creators creates an ADD_TODO action 1`] = `
Array [
  Object {
    "payload": Object {
      "completed": false,
      "id": "testTodoId",
      "text": "This is text todo text.",
    },
    "type": "add_todo",
  },
]
`;

exports[`action creators creates an SET_TODO_LIST_FILTER action 1`] = `
Array [
  Object {
    "payload": Object {
      "todoListFilter": "ALL",
    },
    "type": "set_todo_list_filter",
  },
]
`;

exports[`action creators creates an TOGGLE_TODO_COMPLETION action 1`] = `
Array [
  Object {
    "payload": Object {
      "completed": false,
      "id": "1UpbCPrF6mODLiVE6Uw7-",
      "text": "Sample Todo Text.",
    },
    "type": "toggle_todo_completion",
  },
]
`;
